### DISCUSSION TITLE
How to serve an specific endpoint from another port

### DISCUSSION THREAD

**Discussion by danipl** [rwf2/Rocket#2494] [Questions]
**Use case**

When implementing monitoring capabilities with, for instance, Prometheus, a common way of exposing those metrics is through a specific endpoint where Prometheus pools to scrap those metrics.

A common way to face security policies is to expose that endpoint through a different port, so dealing with security issues becomes easier, just exposing publicly the application port, keeping the other just reachable from the internal network.

**Appcoahes**

1. In the beginning, I tried to find some configuration mechanism to do it, but it is nothing there.
2. Later on, I tried to run two separated Rocket instances from the same process, so they share from the memory the Prometheus resources, but the approach does not work.

I tried... but only starts the first instance launched.
```rust
#[rocket::main]
async fn main() {
    ...
    let rocket_web = rocket::build()
        .manage(create_db_connection())
        .mount("/public", FileServer::from("./webapp/static"))
        .mount("/", routes![...])
        .attach(Session::fairing().with_lifetime(Duration::from_secs(15 * 60)))
        .attach(Template::fairing());
    let rocket_management = rocket::custom(Config {
        port: port.parse::<u16>().unwrap(),
        address: Ipv4Addr::new(0, 0, 0, 0).into(),
        ident: Ident::try_new("Rocket Management").unwrap(),
        ..Config::debug_default()
    }).mount("/", routes![prometheus]);
    rocket_web.launch().await;
    rocket_management.launch().await;
}
```
Thank you.

**Comment by the10thWiz** [comment]
The future returned by launch only resolves (i.e. `.await` returns) when the server shuts down. You should instead use `join!(rocket_web.launch(), rocket_management.launch())`, which will await on both futures simulatiously.
